File: Algo/Leetcode/258AddDigits.cpp
// 能被9除，则为9. 否则余数就是最后的结果
class Solution {
public:
    int addDigits(int num) {
        if(num<10) return num;
        return num%9 == 0?9:num%9;
    }
};

// 更加简化
class Solution {
public:
    int addDigits(int num) {
        return (num-1)%9+1;
    }
};

File: Algo/Leetcode/258AddDigits.py
class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        while num >= 10:
            num = addSum(num)

        return num

def addSum(num):
    return sum([int(n) for n in str(num)])


# Most simple
class Solution(object):
    def addDigits(self, num):
        """
        :type num: int
        :rtype: int
        """
        return (num-1)%9 + 1 if num else 0